code/espurna/sensors/AnalogSensor.h

```diff
@@ -94,8 +115,28 @@ class AnalogSensor : public BaseSensor {
             return sum / _samples;
         }
 
+        //CICM: and proper read should be scalable and thus needs sign
+        //and decimal part
+        double _read() {
+          //Raw measure could also be a class variable with getter so that can
+          //be reported through MQTT, ...
+          unsigned int rawValue;
+          double scaledValue;
+          // Debugging doubles to string
+          //debugSend(("[ANALOG_SENSOR] Started standard read, factor: %s , offset: %s, decimals: %d \n"), String(_factor).c_str(), String(_offset).c_str(), ANALOG_DECIMALS);
+          rawValue = _rawRead();
+          //debugSend(("[ANALOG_SENSOR] Raw read received: %d \n"), rawValue);
+          scaledValue = _factor*rawValue  + _offset;
+          debugSend(("[ANALOG_SENSOR] Scaled value result: %s \n"), String(scaledValue).c_str());
```

Comment:
Use DEBUG_MSG macro instead of debugSend.

Reply:
Should be correct now